operator/src/main/java/org/keycloak/operator/controllers/KeycloakDistConfigurator.java

```diff
@@ -82,34 +83,32 @@ public void configureDistOptions() {
      */
     public void validateOptions(KeycloakStatusBuilder status) {
         assumeFirstClassCitizens(status);
+
+        validateHostname(status);
     }
 
     /* ---------- Configuration of first-class citizen fields ---------- */
 
     public void configureHostname() {
-        var kcContainer = deployment.getSpec().getTemplate().getSpec().getContainers().get(0);
-        var hostname = keycloakCR.getSpec().getHostname();
-        var envVars = kcContainer.getEnv();
-        if (keycloakCR.getSpec().isHostnameDisabled()) {
-            var disableStrictHostname = List.of(
-                    new EnvVarBuilder()
-                            .withName("KC_HOSTNAME_STRICT")
-                            .withValue("false")
-                            .build(),
-                    new EnvVarBuilder()
-                            .withName("KC_HOSTNAME_STRICT_BACKCHANNEL")
-                            .withValue("false")
-                            .build());
+        optionMapper(keycloakCR.getSpec().getHostnameSpec())
+                .mapOption("hostname", HostnameSpec::getHostname)
+                .mapOption("hostname-url", HostnameSpec::getUrl)
+                .mapOption("hostname-port", HostnameSpec::getPort)
+                .mapOption("hostname-admin", HostnameSpec::getAdmin)
+                .mapOption("hostname-admin-url", HostnameSpec::getAdminUrl)
+                .mapOption("hostname-strict", HostnameSpec::isStrict)
+                .mapOption("hostname-strict-backchannel", HostnameSpec::isStrictBackchannel);
+    }
 
-            envVars.addAll(disableStrictHostname);
-        } else {
-            var enabledStrictHostname = List.of(
-                    new EnvVarBuilder()
-                            .withName("KC_HOSTNAME")
-                            .withValue(hostname)
-                            .build());
+    private void validateHostname(KeycloakStatusBuilder status) {
+        var hostnameSpec = keycloakCR.getSpec().getHostnameSpec();
+        if (hostnameSpec != null && hostnameSpec.getPort() != null) {
+            var specifiedPort = hostnameSpec.getPort();
 
-            envVars.addAll(enabledStrictHostname);
+            if (!specifiedPort.equals(Constants.KEYCLOAK_HTTP_PORT) || !specifiedPort.equals(Constants.KEYCLOAK_HTTPS_PORT)) {
+                status.addWarningMessage("It is not possible to change 'hostname-port' at this moment. " +
```

Comment:
I'm not sure if we can so strict about the `hostname-port` because you might have proxy in front of the cluster. I'm saying that based on past issues where there is a proxy chain (e.g.: https://github.com/keycloak/keycloak/issues/13261) and people exposing the server using a port other than the defaults.

For me we should either remove `hostname-port` from the CR or remove this validation.

Reply:
Interesting issue with the proxy chain. Thanks for bringing this up. I'm in to remove `hostname-port` for now, as described below.